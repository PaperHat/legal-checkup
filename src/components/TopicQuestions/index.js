import { map } from 'ramda'
import React from 'react'

import StyledDesktopQuestionButton from '@components/styled/DesktopQuestionButton'

import StyledTopicName from '@components/styled/TopicName'

import StyledTopicQuestions from '@components/styled/TopicQuestions'

import makeQuestionButton from '@wrappers/makeQuestionButton'

export default function TopicQuestions ({ topic, topicQuestions, activeTopic }) {
  console.log('compoent', activeTopic)
  return (
    <StyledTopicQuestions activeTopic={activeTopic}>
      <StyledTopicName>{topic}</StyledTopicName>
      <div>
        {map(({ id, index }) => {
          const QuestionButton = makeQuestionButton(StyledDesktopQuestionButton)

          return (
            <QuestionButton key={id} questionIndex={index}>
              {index + 1}
            </QuestionButton>
          )
        }, topicQuestions)}
      </div>
    </StyledTopicQuestions>
  )
}
