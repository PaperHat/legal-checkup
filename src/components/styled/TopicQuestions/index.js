import styled from 'styled-components'

const TopicQuestions = styled.div`
  font-weight: ${({ activeTopic }) => (activeTopic ? 'bold' : 'normal')};

  background-color: #fff;
  display: inline-block;
`

TopicQuestions.displayName = 'StyledTopicQuestions'

export default TopicQuestions
