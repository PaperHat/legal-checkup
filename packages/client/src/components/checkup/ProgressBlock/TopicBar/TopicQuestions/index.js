import styled from 'styled-components'

const TopicQuestions = styled.div`
  background-color: #ededed;
  font-weight: ${({ activeTopic }) => (activeTopic ? 'bold' : 'normal')};
  margin-top: -55px;
  margin-bottom: 20px;
  margin-left: -5px;
  margin-right: 5px;
  background-color: transparent;
  display: inline-block;
  color: ${({ activeTopic }) => (activeTopic ? '#b62645' : '#ededed')};
`

TopicQuestions.displayName = 'StyledTopicQuestions'

export default TopicQuestions
