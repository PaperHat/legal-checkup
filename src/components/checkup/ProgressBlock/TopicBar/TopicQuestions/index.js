import styled from 'styled-components'

const TopicQuestions = styled.div`
  background-color: #ededed;
  font-weight: ${({ activeTopic }) => (activeTopic ? 'bold' : 'normal')};
  margin-top: -55px;
  margin-bottom: 20px;
  margin-left: -5px;
  margin-right: 5px;
  display: inline-block;
  color: ${({ activeTopic }) => (activeTopic ? '#b62645' : '#a1a1a1')};
`

TopicQuestions.displayName = 'StyledTopicQuestions'

export default TopicQuestions
