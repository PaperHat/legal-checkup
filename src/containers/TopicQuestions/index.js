import { connect } from 'react-redux'

import { makeGetTopicQuestions } from '@state/selectors'
import TopicQuestions from '@components/TopicQuestions'

function makeMapStateToProps () {
  const getTopicQuestions = makeGetTopicQuestions()
  return function mapStateToProps (state, { topic }) {
    return {
      topicQuestions: getTopicQuestions(state, { topic })
    }
  }
}

export default connect(makeMapStateToProps)(TopicQuestions)
