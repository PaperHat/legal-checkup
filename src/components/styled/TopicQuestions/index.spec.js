import React from 'react'
import { shallow } from 'enzyme'

import TopicQuestions from '.'

describe('components:styled:TopicQuestions', () => {
  it('renders the TopicQuestions and CSS properties properly', () => {
    expect(
      toJson(shallow(<TopicQuestions>Topic questions</TopicQuestions>))
    ).toMatchSnapshot()
  })
  it('renders the TopicQuestions and CSS properties properly when activeTopic', () => {
    expect(
      toJson(
        shallow(<TopicQuestions activeTopic>Topic questions</TopicQuestions>)
      )
    ).toMatchSnapshot()
  })
})
