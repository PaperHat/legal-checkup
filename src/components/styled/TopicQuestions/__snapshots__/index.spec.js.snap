// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:styled:TopicQuestions renders the TopicQuestions and CSS properties properly 1`] = `
.c0 {
  background-color: #ededed;
  font-weight: #c0c0c0;
  margin-top: -150px;
  margin-bottom: 20px;
  margin-left: -5px;
  margin-right: 5px;
  background-color: #fff;
  display: inline-block;
}

<div
  className="c0"
>
  Topic questions
</div>
`;

exports[`components:styled:TopicQuestions renders the TopicQuestions and CSS properties properly when activeTopic 1`] = `
.c0 {
  background-color: #ededed;
  font-weight: bold;
  margin-top: -150px;
  margin-bottom: 20px;
  margin-left: -5px;
  margin-right: 5px;
  background-color: #fff;
  display: inline-block;
}

<div
  className="c0"
>
  Topic questions
</div>
`;
